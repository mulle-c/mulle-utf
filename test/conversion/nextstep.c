#include <mulle-utf/mulle-utf.h>

#include <stdio.h>
#include <stdlib.h>



int  main()
{
   char            nextstep[ 256];
   int             n;
   size_t    m;
   char            buf[ 256 * 4] = { 0 };
   char            *end;
   char            *s;
   char            *t;
   int             i;
   size_t    off;

   for( n = 0, i = 32; i < 254; i++, n++)
      nextstep[ n] = i;

   end = _mulle_nextstep_convert_to_utf8( nextstep, n, buf);

   off = 32;
   s   = buf;
   for(;;)
   {
      m = 16;
      t = mulle_utf8_skiputf32( s, &m);
      if( m != 16)
         break;

      printf( "%02x: %.*s\n", off, (int) (t - s), s);
      s    = t;
      off += 16;
   }

   if( end != s)
      printf( "%02x: %.*s\n", off, (int) (end - s), s);

   return( 0);
}

