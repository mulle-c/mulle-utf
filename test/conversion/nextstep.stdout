20:  !"#$%&'()*+,-./
30: 0123456789:;<=>?
40: @ABCDEFGHIJKLMNO
50: PQRSTUVWXYZ[\]^_
60: `abcdefghijklmno
70: pqrstuvwxyz{|}~
80:  ÀÁÂÃÄÅÇÈÉÊËÌÍÎÏ
90: ÐÑÒÓÔÕÖÙÚÛÜÝÞµ×÷
a0: ©¡¢£⁄¥ƒ§¤’“«‹›ﬁﬂ
b0: ®–†‡·¦¶∙‚„”»⋯‰¬¿
c0: ¹̀´̂̃¯˘˙¨²°¸³˝˛ˇ
d0: ̵±¼½¾àáâãäåçèéêë
e0: ìÆíªîïðñŁØŒºòóôõ
f0: öæùúûıüýłøœßþÿ
